blocktx/store/sql/InsertBlockTransactions.go

```diff
@@ -42,7 +42,7 @@ func (s *SQL) InsertBlockTransactions(ctx context.Context, blockId uint64, trans
 	for pos, tx := range transactions {
 		var txid uint64
 
-		if err = qTx.QueryRowContext(ctx, tx.Hash).Scan(&txid); err != nil {
+		if err = qTx.QueryRowContext(ctx, tx.Hash, merklePaths[pos]).Scan(&txid); err != nil {
```

Comment:
It's not your code but, this block looks very complicated to me. I think it can be simplified to this:

```
		err = qTx.QueryRowContext(ctx, tx.Hash, merklePaths[pos]).Scan(&txid)
		if err != nil {
			if err != sql.ErrNoRows {
				return err
			}

			err = s.db.QueryRowContext(ctx, `
					SELECT id
					FROM transactions
					WHERE hash = $1
				`, tx.Hash).Scan(&txid)
			if err != nil {
				return err
			}
		}
```